Vagrantfile

```diff
@@ -106,6 +106,7 @@ Vagrant.configure('2') do |config|
       export PATH=$PATH:/opt/chefdk/bin:/opt/chefdk/embedded/bin
       nohup chef-zero -H 0.0.0.0 -p 4000 2>&1 > /dev/null &
       cd /vagrant/vagrant_repo
+      gem install bundler:2.0.2
```

Comment:
How come this was added?  It doesn't appear to be necessary, I commented it out and the box still built fine.

Reply:
It does complain about bundler not being up to date if I take this line out.
```
chef: Running provisioner: shell...
    chef: Running: inline script
    chef: killed chef-zero
    chef: kill: usage: kill [-s sigspec | -n signum | -sigspec] pid | jobspec ... or kill -l [sigspec]
==> chef: Running provisioner: shell...
    chef: Running: inline script
    chef: /opt/chefdk/embedded/lib/ruby/2.6.0/rubygems/dependency.rb:313:in `to_specs': Could not find 'bundler' (2.0.2) required by your /vagrant/Gemfile.lock. (Gem::MissingSpecVersionError)
    chef: To update to the latest version installed on your system, run `bundle update --bundler`.
    chef: To install the missing version, run `gem install bundler:2.0.2`
    chef: Checked in 'GEM_PATH=/home/vagrant/.chefdk/gem/ruby/2.6.0:/opt/chefdk/embedded/lib/ruby/gems/2.6.0', execute `gem env` for more information
    chef: 	from /opt/chefdk/embedded/lib/ruby/2.6.0/rubygems/specification.rb:1449:in `block in activate_dependencies'
    chef: 	from /opt/chefdk/embedded/lib/ruby/2.6.0/rubygems/specification.rb:1438:in `each'
    chef: 	from /opt/chefdk/embedded/lib/ruby/2.6.0/rubygems/specification.rb:1438:in `activate_dependencies'
    chef: 	from /opt/chefdk/embedded/lib/ruby/2.6.0/rubygems/specification.rb:1420:in `activate'
    chef: 	from /opt/chefdk/embedded/lib/ruby/2.6.0/rubygems/specification.rb:1452:in `block in activate_dependencies'
    chef: 	from /opt/chefdk/embedded/lib/ruby/2.6.0/rubygems/specification.rb:1438:in `each'
    chef: 	from /opt/chefdk/embedded/lib/ruby/2.6.0/rubygems/specification.rb:1438:in `activate_dependencies'
    chef: 	from /opt/chefdk/embedded/lib/ruby/2.6.0/rubygems/specification.rb:1420:in `activate'
    chef: 	from /opt/chefdk/embedded/lib/ruby/2.6.0/rubygems/core_ext/kernel_gem.rb:68:in `block in gem'
    chef: 	from /opt/chefdk/embedded/lib/ruby/2.6.0/rubygems/core_ext/kernel_gem.rb:67:in `synchronize'
    chef: 	from /opt/chefdk/embedded/lib/ruby/2.6.0/rubygems/core_ext/kernel_gem.rb:67:in `gem'
    chef: 	from /usr/bin/knife:54:in `<main>'
    chef: /opt/chefdk/embedded/lib/ruby/2.6.0/rubygems/dependency.rb:313:in `to_specs': Could not find 'bundler' (2.0.2) required by your /vagrant/Gemfile.lock. (Gem::MissingSpecVersionError)
    chef: To update to the latest version installed on your system, run `bundle update --bundler`.
    chef: To install the missing version, run `gem install bundler:2.0.2`
    chef: Checked in 'GEM_PATH=/home/vagrant/.chefdk/gem/ruby/2.6.0:/opt/chefdk/embedded/lib/ruby/gems/2.6.0', execute `gem env` for more information
    chef: 	from /opt/chefdk/embedded/lib/ruby/2.6.0/rubygems/specification.rb:1449:in `block in activate_dependencies'
    chef: 	from /opt/chefdk/embedded/lib/ruby/2.6.0/rubygems/specification.rb:1438:in `each'
    chef: 	from /opt/chefdk/embedded/lib/ruby/2.6.0/rubygems/specification.rb:1438:in `activate_dependencies'
    chef: 	from /opt/chefdk/embedded/lib/ruby/2.6.0/rubygems/specification.rb:1420:in `activate'
    chef: 	from /opt/chefdk/embedded/lib/ruby/2.6.0/rubygems/specification.rb:1452:in `block in activate_dependencies'
    chef: 	from /opt/chefdk/embedded/lib/ruby/2.6.0/rubygems/specification.rb:1438:in `each'
    chef: 	from /opt/chefdk/embedded/lib/ruby/2.6.0/rubygems/specification.rb:1438:in `activate_dependencies'
    chef: 	from /opt/chefdk/embedded/lib/ruby/2.6.0/rubygems/specification.rb:1420:in `activate'
    chef: 	from /opt/chefdk/embedded/lib/ruby/2.6.0/rubygems/core_ext/kernel_gem.rb:68:in `block in gem'
    chef: 	from /opt/chefdk/embedded/lib/ruby/2.6.0/rubygems/core_ext/kernel_gem.rb:67:in `synchronize'
    chef: 	from /opt/chefdk/embedded/lib/ruby/2.6.0/rubygems/core_ext/kernel_gem.rb:67:in `gem'
    chef: 	from /usr/bin/berks:54:in `<main>'
    chef: /opt/chefdk/embedded/lib/ruby/2.6.0/rubygems/dependency.rb:313:in `to_specs': Could not find 'bundler' (2.0.2) required by your /vagrant/Gemfile.lock. (Gem::MissingSpecVersionError)
    chef: To update to the latest version installed on your system, run `bundle update --bundler`.
    chef: To install the missing version, run `gem install bundler:2.0.2`
    chef: Checked in 'GEM_PATH=/home/vagrant/.chefdk/gem/ruby/2.6.0:/opt/chefdk/embedded/lib/ruby/gems/2.6.0', execute `gem env` for more information
    chef: 	from /opt/chefdk/embedded/lib/ruby/2.6.0/rubygems/specification.rb:1449:in `block in activate_dependencies'
    chef: 	from /opt/chefdk/embedded/lib/ruby/2.6.0/rubygems/specification.rb:1438:in `each'
    chef: 	from /opt/chefdk/embedded/lib/ruby/2.6.0/rubygems/specification.rb:1438:in `activate_dependencies'
    chef: 	from /opt/chefdk/embedded/lib/ruby/2.6.0/rubygems/specification.rb:1420:in `activate'
    chef: 	from /opt/chefdk/embedded/lib/ruby/2.6.0/rubygems/specification.rb:1452:in `block in activate_dependencies'
    chef: 	from /opt/chefdk/embedded/lib/ruby/2.6.0/rubygems/specification.rb:1438:in `each'
    chef: 	from /opt/chefdk/embedded/lib/ruby/2.6.0/rubygems/specification.rb:1438:in `activate_dependencies'
    chef: 	from /opt/chefdk/embedded/lib/ruby/2.6.0/rubygems/specification.rb:1420:in `activate'
    chef: 	from /opt/chefdk/embedded/lib/ruby/2.6.0/rubygems/core_ext/kernel_gem.rb:68:in `block in gem'
    chef: 	from /opt/chefdk/embedded/lib/ruby/2.6.0/rubygems/core_ext/kernel_gem.rb:67:in `synchronize'
    chef: 	from /opt/chefdk/embedded/lib/ruby/2.6.0/rubygems/core_ext/kernel_gem.rb:67:in `gem'
    chef: 	from /usr/bin/berks:54:in `<main>'
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
```

Reply:
So it seems that only happens if you have bundler 2.x installed locally when you ran bundle install/update. (Note: Ruby 2.6.5 comes with bundler 1.17.2).  For some reason, the knife command loads up bundler and it somehow finds the Gemfile.lock in the mounted /vagrant directory.  If that Gemfile.lock was created with 2.x, it barfs if you only have a bundler 1.x version installed.  I can't seem to find any way around that, so I suppose it's fine.  Although it may be better to just `gem install bundler` to grab the latest version instead of pulling 2.0.2 specifically.

Reply:
@gravesb I agree. I updated it. All other concerns are addressed. Please give a look once more. Thanks!